Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -535,12 +535,21 @@ public partial class Translator {
       List<Variable> inParams = Boogie.Formal.StripWhereClauses(proc.InParams);
       List<Variable> outParams = Boogie.Formal.StripWhereClauses(proc.OutParams);
 
-      BoogieStmtListBuilder builder = new BoogieStmtListBuilder(this, options);
+      var builder = new BoogieStmtListBuilder(this, options);
+      var builderInitializationArea = new BoogieStmtListBuilder(this, options);
       builder.Add(new CommentCmd("AddMethodImpl: " + m + ", " + proc));
       var etran = new ExpressionTranslator(this, predef, m.tok);
+      // Only do reads checks for methods, not lemmas
+      // (which aren't allowed to declare frames and don't check reads and writes against them).
+      // Also don't do any reads checks if the reads clause is *,
+      // since all the checks will be vacuously true
```

Comment:
```suggestion
      // since all the checks will be trivially true
```

Vacuously = True because false is assumed
Trivially = True because it's an axiom that something belongs to everything